src/control/lib/control/server_meta_test.go

```diff
@@ -640,11 +655,80 @@ func TestControl_SmdManage(t *testing.T) {
 			},
 			expErr: errors.New("> 1 host"),
 		},
+		// set-faulty API calls do not return SMD info.
 		"set-faulty": {
 			req: &SmdManageReq{
 				Operation: SetFaultyOp,
 				IDs:       test.MockUUID(1),
 			},
+			mic: newMockInvokerWRankResps(&ctlpb.SmdManageResp_RankResp{
+				Rank:    0,
+				Results: []*ctlpb.SmdManageResp_Result{{}},
+			}),
+			expResp: &SmdResp{
+				HostStorage: mockSmdQueryMap(t, &mockSmdResp{Hosts: "host-0"}),
+			},
+		},
+		"set-faulty; drpc failure": {
```

Comment:
nit - with the mock invoker, this would be gRPC not dRPC.

Reply:
The daos.Busy failure is coming from dRPC response which is contained in the rank result (SmdManageResp_Result). The outer response (SmdManageResp) is what will be transmitted over gRPC. The error is encapsullated within the inner rank result so I think the comment is valid.

Reply:
Ah, I see. I guess my confusion is none of the drpc logic is touched, the test is simulating a bad engine response (which incidentally comes up through drpc). Maybe "bad status" would be a clearer comment?

Reply:
updated comment to read rank failure